### DISCUSSION TITLE
V7: TypeError: props.render is not a function at Controller (webpack-internal:///./node_modules/react-hook-form/dist/index.esm.js:1645:37)

### DISCUSSION THREAD

**Discussion by saurabhpabhare** [react-hook-form/react-hook-form#4598] [Q&A]
I am creating a form with a Material UI Autocomplete field with checkbox.

Following is the link for my CSB -> [https://codesandbox.io/s/sweet-wright-2ghnm?file=/src/index.js](https://codesandbox.io/s/sweet-wright-2ghnm?file=/src/index.js)

It seems to work properly on CSB, but on my local with same package.json it is throwing following error.

<img width="1553" alt="Screenshot 2021-04-03 at 1 31 14 PM" src="https://user-images.githubusercontent.com/30390186/113472441-e003c400-9480-11eb-96c5-a64b21bf4e90.png">

<img width="555" alt="Screenshot 2021-04-03 at 1 33 16 PM" src="https://user-images.githubusercontent.com/30390186/113472494-3b35b680-9481-11eb-9282-cb940232f034.png">

<img width="751" alt="Screenshot 2021-04-03 at 1 32 55 PM" src="https://user-images.githubusercontent.com/30390186/113472497-3ffa6a80-9481-11eb-873b-a3da98f286f9.png">

I have been at this for a long while now, so any pointers would be really great help.

Thank you very much!

**Comment by jorisre** [comment]
That's strange, you're using the same code in your project and in the codesandbox ?
Do you get that error after updating to React Hook Form 7 ?

**Reply by saurabhpabhare** [reply]
Yeah I'm using same code in my project. Getting this error after updating it to V7.

**Reply by bluebill1049** [reply]
after update to v7? there are breaking changes, maybe you should stay with v6.

**Reply by saurabhpabhare** [reply]
With V6 I'm getting this error now. 😞 

<img width="729" alt="Screenshot 2021-04-03 at 2 40 50 PM" src="https://user-images.githubusercontent.com/30390186/113473983-991acc00-948a-11eb-9a01-bdf777a2f922.png">

**Reply by saurabhpabhare** [reply]
@bluebill1049 there might be breaking changes in V7 but I am wondering, how come it's not breaking in [https://codesandbox.io/s/sweet-wright-2ghnm?file=/src/index.js:1210-1219](url)

**Reply by bluebill1049** [reply]
it's not working tho, you should console.log `controllerProps`

**Reply by saurabhpabhare** [reply]
I checked. It's coming. It valid data is also coming on submit of RHF. Please refer attached screenshot. 

<img width="1791" alt="Screenshot 2021-04-03 at 3 10 52 PM" src="https://user-images.githubusercontent.com/30390186/113474671-fd3f8f00-948e-11eb-82d9-83bc85b7f652.png">

**Comment by bluebill1049** [comment]
have you tried to remove the entire node_modules and do a npm/yarn?

**Reply by saurabhpabhare** [reply]
Yes. I removed node_modules and package-lock.json and ran npm install again couple of times. Still getting the same error. It's breaking in all the forms where I'm using render prop of Controller.

**Comment by jorisre** [comment]
@saurabhpabhare Can you share a github repo with the problem ?
